ts/parsers/slideParser.ts

```diff
@@ -126,14 +126,14 @@ export default class SlideParser {
         return { groupedShapes, groupedImages };
     }
 
-    public static async getSlideElements(PPTElementParser: PowerpointElementParser, slideNumber): Promise<any[]> {
+    public static async getSlideElements(PPTElementParser: PowerpointElementParser, slideNumber, pptFilePath: string): Promise<any[]> {
         //Get all of Slide Shapes and Elements
-        const slideAttributes = await ZipHandler.parseSlideAttributes(format("ppt/slides/slide{0}.xml", slideNumber));
+        const slideAttributes = await FileHandler.parseSlideAttributes(join(pptFilePath, format("ppt/slides/slide{0}.xml", slideNumber)));
         //Contains references to links,images and etc on a Slide
-        const slideRelations = await ZipHandler.parseSlideAttributes(
-            format("ppt/slides/_rels/slide{0}.xml.rels", slideNumber)
+        const slideRelations = await FileHandler.parseSlideAttributes(
+            join(pptFilePath, format("ppt/slides/_rels/slide{0}.xml.rels", slideNumber))
         );
-        const { slideMasterTables, slideLayoutTables } = await this.getSlideLayout(slideRelations);
+        const { slideMasterTables, slideLayoutTables } = await this.getSlideLayout(slideRelations, pptFilePath);
         const slideData = slideAttributes["p:sld"]["p:cSld"];
```

Comment:
I think all of this should be in try catch as well .. in case slide{n}.xml is not found .. especially now you rely on the renderer to give you the file path, it could lie to you :)

Reply:
done